### ISSUE TITLE
Bug: IPropertyHandler not being called

### ISSUE THREAD

**Issue by bcssov** [mikependon/RepoDB#514]
I'm referencing this issue #437 and, yes, I'm aware of the pitfalls of doing string to Guid conversion. DB design is not under my control so I can only consume it. 

First of all I've switched to MDS in the sample and have discovered another unrelated issue as reported here #513.

Now the default sample works but in a scenario where you use map attribute the property handler is not being called. As I mentioned the db design is not under my control I am merely a consumer (third party). So consider the following example.
- Create the following table:
```
connection.ExecuteNonQuery(@"CREATE TABLE IF NOT EXISTS demoUpsertObj
	(
		id char(36) PRIMARY KEY,
		entityId INTEGER,
		name TEXT NOT NULL,
		validFrom Datetime,
		validTo  datetime,
		deleted byte
	);");
```
- Decorate the model as follows:
```
    [Map("demoUpsertObj")]
    public class DemoUpsertObj //: IUpsertEntity
    {
        [Map("id"), Primary, PropertyHandler(typeof(StringToGuidPropertyHandler))]
        public Guid Id { get; set; }
        [Map("entityId")]
        public long EntityId { get; set; }
        [Map("validFrom")]
        public DateTime ValidFrom { get; set; }
        [Map("validTo")]
        public DateTime ValidTo { get; set; }
        [Map("deleted")]
        public byte Deleted { get; set; }
        [Map("name")]
        public string Name { get; set; }
        public DemoUpsertObj() { }
        public DemoUpsertObj(string name)
        {
            Name = name;
        }
    }
```
- Perform the query:
``` var result = connection.Query<DemoUpsertObj>(p => p.Id == recordId).FirstOrDefault();```

No results will be returned. Remove the map attribute or just lowercase the property and a match will be returned.

Modified sample project: 
[RepoDbIssue437.zip](https://github.com/mikependon/RepoDb/files/5060160/RepoDbIssue437.zip)

**Comment by mikependon** [comment]
Not yet verified on our end, but we will get back to you soon. Do you have timeline when do you need the fix here?

**Comment by bcssov** [comment]
No timelines, I workaround this by lower-casing property names at the moment.

**Comment by mikependon** [comment]
@bcssov - hey, I can't replicate this issue. Are you using the latest version of RepoDB here? Thanks

**Comment by mikependon** [comment]
The fix is now available at [RepoDb v1.12.0-beta4](https://www.nuget.org/packages/RepoDb/1.12.0-beta4) and [RepoDb.SqlServer v1.1.0-beta2](https://www.nuget.org/packages/RepoDb.SqlServer/1.1.0-beta2).

```csharp
> Install-Package RepoDb -version 1.12.0-beta4
> Install-Package RepoDb.SqlServer -version 1.1.0-beta2
```

Please do let us know if you still have questions. Thanks!
